### PR TITLE
Reader: Ensure tag follow and unfollow button is fully clickable

### LINKED ISSUE #33762: Reader: following tag button has inconsistent hit area
#### Steps to reproduce
1. Head to https://wordpress.com/tag/cheese
2. Interact with the tag follow and unfollow button at the top right of the screen
3. Tested in Chrome, the button is hard to press, the hit area is small and inconsistent

#### What I expected

It to be east to follow and unfollow a tag with the mouse

#### What happened instead

I found it hard to follow a tag

#### Browser / OS version

Mac OS X 10.14
Chrome 74.0

#### Screenshot / Video

Screenshot 2019-06-07 at 20 21 04

#### Context / Source
#user-report

### PR DESCRIPTION
**Pull request by scottsweb**
#### Changes proposed in this Pull Request

* Increase `z-index` of tag follow button in the reader to ensure it remains clickable when the 'Back' button is visible.

#### Testing instructions

1. Head to https://wordpress.com/tag/cheese
2. Interact with the tag follow and unfollow button at the top right of the screen
3. Notice not all of it is clickable when the 'back' button is present (as per screenshot below)
4. Pull down this PR
5. Test again, it should be easier to click

Screenshot 2019-06-11 at 11 40 19

Fixes #33762

### CODE REVIEW

File: client/reader/tag-stream/style.scss [hunk 1]
```diff
@@ -14,6 +14,7 @@
 }
 
 .tag-stream__header .follow-button {
+	z-index: 1;
```
Comment (torres126) [inline client/reader/tag-stream/style.scss]: It's probably better to use the scss z-index function here: https://wpcalypso.wordpress.com/devdocs/docs/coding-guidelines/css.md#z-index

**Comment by scottsweb** [comment]
Thanks for the suggestions, please let me know if this is the correct approach now.

File: client/reader/tag-stream/style.scss [hunk 2]
```diff
@@ -14,7 +14,7 @@
 }
 
 .tag-stream__header .follow-button {
-	z-index: 1;
+	z-index: z-index( '.tag-stream__header-follow', 'button.button.follow-button.has-icon..is-following' );
```
Comment (jancavan) [inline client/reader/tag-stream/style.scss]: Not sure why there are two periods here? @scottsweb

Comment (jancavan) [reply client/reader/tag-stream/style.scss]: `has-icon..is-following`

Comment (scottsweb) [reply client/reader/tag-stream/style.scss]: I used the Chrome extension to generate these and this is what it gave me

**Comment by bluefuton** [comment]
@scottsweb @jancavan I've tried to simplify it a bit - looks like the correct `z-index` is still being applied:

Screen Shot 2019-06-13 at 15 28 10

**Comment by scottsweb** [comment]
Thanks @bluefuton, this looks good to me. The simpler references are much better. I didn't appreciate that we could simplify them in this way.

**Comment by jancavan** [comment]
This is working now 👍
